src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs

```diff
@@ -1110,10 +1110,15 @@ private void LoadExtensions(Assembly assembly, bool throwOnError)
         /// </remarks>
         private static bool CanContainExtensions(Assembly assembly)
         {
+            // We used to have two separate strong name keys for product and test binaries and Arcade
+            // does not support that anymore (and that did not accept our PR for adding the additional
+            // signed key and using two strong name keys without the PR). If they ever decide to support
+            // this scenario we can remove the regex and go back to checking just the public key and startswith.
             if (assembly.FullName.StartsWith("Microsoft.ML.", StringComparison.Ordinal)
-                && HasMLNetPublicKey(assembly))
+               && !Regex.IsMatch(assembly.FullName, @"Microsoft\.ML.*\.Tests.*")
```

Comment:
Instead of changing product code for this (which is basically slowing down the happy path), we can just change the 1 test assembly that tests this functionality. Either change the test assembly to:

1. Not start with `Microsoft.ML.`
2. Change its public key to something other than what the product uses. This can easily be done by setting the `StrongNameKeyId` property on the test project to something other than `Open`.

Reply:
When I change the `StrongNameKeyId` I get an error because of the `InternalsVisibleTo` in the `AssemblyInfo`. I can make the change as long as I can find one of the actual values for the public keys found [here](https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/StrongName.targets), but I am not sure where to find them. Do you know by chance?

Reply:
The way I typically find that information is by using the [`sn.exe` tool](https://docs.microsoft.com/en-us/dotnet/framework/tools/sn-exe-strong-name-tool) in the Visual Studio Developer Command Prompt. I believe the options you want are:

`sn.exe -Tp <path to .dll>`

Another option is to pull out the extensions used in the tests to a separate assembly.

Reply:
@eerhardt The correct fix was in [this PR](https://github.com/dotnet/arcade/pull/6203) that didn't get merged to Arcade. 
Arcade does the build and signing together, so it needs the keys as part of the environment. I couldn't find any way to override it from outside Arcade (even after consultation with the Arcade folks). Checking the test signing key into Arcade is the only fix that worked and that PR not accepted. So we are now signing all binaries with the same key. Please let us know if you have a workaround.

Reply:
Arcade already has more than one public key that can be used. You can just pick a different one for that one test assembly.

Or another option is to put the types that have extensions in them in a separate assembly that doesn't start with `Microsoft.ML.`.

Reply:
[Here]( https://github.com/dotnet/machinelearning/pull/5475) is the PR that fixes the issue @eerhardt, I have requested your review on it. It is adding back in the test cert we were using before since we already had all the info for that cert.